ttnn/cpp/ttnn/operations/data_movement/tilize_with_val_padding/tilize_with_val_padding.cpp

```diff
@@ -7,11 +7,78 @@
 
 namespace ttnn::operations::data_movement {
 
+using OwnedTilizeValArgs = std::tuple<ttnn::Tensor>;
+using BaseTilizeValType = std::function<ttnn::Tensor(const ttnn::Tensor&)>;
+
+using MassagedTilizeVal = MassagedOperation<ttnn::Tensor, const ttnn::Tensor&>;
+using MassagedTilizeValParams = MassagedOperationParams<ttnn::Tensor, const ttnn::Tensor&>;
+
+uint TILE_WIDTH = 32;
+uint TILE_HEIGHT = 32;
+ttnn::Shape update_original_shape(ttnn::Shape& original) {
+    std::vector<uint32_t> update_original(original.rank());
+    if (original[original.rank() - 2] % TILE_HEIGHT != 0) {
+        update_original[original.rank() - 2] = (original[original.rank() - 2] / TILE_HEIGHT + 1) * TILE_HEIGHT;
+        for (int i = 0; i < original.rank(); i++) {
+            if (i != original.rank() - 2) {
+                update_original[i] = original[i];
+            }
+        }
+        return ttnn::Shape(tt::tt_metal::LegacyShape(update_original));
+    }
+
+    else if (original[original.rank() - 1] % TILE_WIDTH != 0) {
+        update_original[original.rank() - 1] = (original[original.rank() - 1] / TILE_WIDTH + 1) * TILE_WIDTH;
+        for (int i = 0; i < original.rank(); i++) {
+            if (i != original.rank() - 1) {
+                update_original[i] = original[i];
+            }
+        }
+        return ttnn::Shape(tt::tt_metal::LegacyShape(update_original));
+    }
+    return original;
+}
+
+MassagedTilizeVal build_ndiml_tilize_val(BaseTilizeValType base_tilize) {
+    auto original_shape = std::make_shared<ttnn::Shape>(ttnn::Shape{});
+    return MassagedTilizeVal(MassagedTilizeValParams{
+        .predicate = [](const ttnn::Tensor& input_tensor) -> bool { return input_tensor.get_shape().rank() > 4; },
+        .pre_transform = [=](const ttnn::Tensor& input_tensor) -> OwnedTilizeValArgs {
+            *original_shape = input_tensor.get_shape();
+            ttnn::Tensor squeezed_tensor = squeeze_to_le_4D(input_tensor);
+            auto print_shape = squeezed_tensor.get_shape();
+            return std::make_tuple(squeezed_tensor);
+        },
+        .post_transform = [=](const ttnn::Tensor& output) -> ttnn::Tensor {
+            auto unsqueezed_tensor = ttnn::reshape(output, update_original_shape(*original_shape));
+            return unsqueezed_tensor;
+        },
+        .operation = std::move(base_tilize)});
+}
+
+tt::tt_metal::LegacyShape squeeze_output_shape(tt::tt_metal::LegacyShape output_shape) {
+    if (output_shape.rank() > 4) {
+        std::array<uint32_t, 4> output_shape_4d;
+        output_shape_4d[0] = 1;
+        int extra_rank = output_shape.rank() - 4;
+        for (int i = extra_rank; i >= 0; i--) {
+            output_shape_4d[0] *= output_shape[i];
+        }
+        output_shape_4d[1] = output_shape[1 + extra_rank];
+        output_shape_4d[2] = output_shape[2 + extra_rank];
+        output_shape_4d[3] = output_shape[3 + extra_rank];
+        return tt::tt_metal::LegacyShape(output_shape_4d);
```

Comment:
Can we see if we can use Simple Shape? I think @jvegaTT did that recently, consult with him if you are having trouble.